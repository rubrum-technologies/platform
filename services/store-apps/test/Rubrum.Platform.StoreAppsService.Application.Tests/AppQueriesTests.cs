using CookieCrumble;
using HotChocolate.Types.Relay;
using Rubrum.Platform.StoreAppsService.Apps;
using Shouldly;
using Xunit;
using static Rubrum.Platform.StoreAppsService.AppTestConstants;

namespace Rubrum.Platform.StoreAppsService;

public class AppQueriesTests : StoreAppsServiceApplicationGraphqlTestBase
{
    private readonly INodeIdSerializer _idSerializer;

    public AppQueriesTests()
    {
        _idSerializer = GetRequiredService<INodeIdSerializer>();
    }

    [Fact]
    public async Task GetAppByIdTest()
    {
        var appId = _idSerializer.Format(nameof(App), TestAppId);

        await using var result = await ExecuteRequestAsync(b => b.SetDocument(
        $$"""
          query {
              appById(id: "{{appId}}") {
                  name
                  ownerId
                  version {
                    major
                    minor
                    patch
                  }
                  enabled
              }
          }
          """));

        result.ShouldNotBeNull();
        result.MatchSnapshot();
    }

}
