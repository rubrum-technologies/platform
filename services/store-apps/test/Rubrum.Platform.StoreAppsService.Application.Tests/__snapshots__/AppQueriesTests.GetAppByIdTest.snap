{
  "data": {
    "appById": {
      "name": "Тестовое приложение",
      "ownerId": "VXNlcjoAAAAAAAAAAAAAAAAAAAAA",
      "version": {
        "major": 1,
        "minor": 0,
        "patch": 0
      },
      "enabled": true
    }
  }
}
